### PR TITLE
feat(codepipeline): make the default CodePipeline Bucket have an encryption key

### PR DESCRIPTION
**Pull request by skinny85**
This is a prerequisite for having a nice cross-account experience
(see #1924).

----

### Pull Request Checklist

* [x] Testing
  - Unit test added (prefer not to modify an existing test, otherwise, it's probably a breaking change)
  - __CLI change?:__ coordinate update of integration tests with team
  - __cdk-init template change?:__ coordinated update of integration tests with team
* [x] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
* [x] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [x] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts [hunk 1]
```diff
@@ -159,7 +160,10 @@ export class Pipeline extends cdk.Construct implements IPipeline {
     // If a bucket has been provided, use it - otherwise, create a bucket.
     let propsBucket = props.artifactBucket;
     if (!propsBucket) {
+      const encryptionKey = new kms.EncryptionKey(this, 'ArtifactsBucketEncryptionKey');
```
Comment (eladb) [inline packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts]: Feels like something we should allow users to opt out from and also supply their own key.

Comment (skinny85) [reply packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts]: We already allow it - the customer can supply their own `artifactBucket` in `props` (notice that we check `if (!propsBucket)`, and don't do any defaulting in that case).

**Comment by skinny85** [comment]
Updated the `decdk` snapshot.

File: packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts (hunk 1 above)
Comment (eladb) [reply packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts]: My comment was about the key, not about the bucket, but I guess you are right. If they want to customize the key, they can just pass in a custom bucket. Make sure to mention in the `artifactBucket` props that the bucket is encrypted.

**Pull request merged**